src/lib/flight_tasks/tasks/Utility/Sticks.cpp

```diff
@@ -44,25 +44,23 @@ Sticks::Sticks(ModuleParams *parent) :
 	ModuleParams(parent)
 {};
 
-bool Sticks::checkAndSetStickInputs(hrt_abstime now)
+bool Sticks::checkAndSetStickInputs()
 {
-	_sub_manual_control_setpoint.update();
-
-	hrt_abstime rc_timeout = (_param_com_rc_loss_t.get() * 1.5f) * 1_s;
-
 	// Sticks are rescaled linearly and exponentially to [-1,1]
-	if ((now - _sub_manual_control_setpoint.get().timestamp) < rc_timeout) {
```

Comment:
We can't have multiple parts of the system checking the same condition differently. I think it could be reasonable that the consumer decides on the validity and reports that back up, but not both. If there's something that slips through here we need to fix it holistically.

If you really want to eliminate the possibility of using stale stick data then the way to do that is gate the processing behind the manual_control_setpoint update. New manual_control_setpoint => new setpoint.

Reply:
If you still want to have a check for now we can use `vehicle_status.rc_signal_lost`.

Reply:
https://github.com/PX4/PX4-Autopilot/pull/16270/commits/3be7a486fbe0067699d41e9db543ddca7a8e7236

Reply:
> We can't have multiple parts of the system checking the same condition differently.

Yes, you're right.

> I think it could be reasonable that the consumer decides on the validity and reports that back up,

That's what I'd prefer to be at least supported from a scalability standpoint.

> gate the processing behind the manual_control_setpoint update

That's what you do now. The trajectory needs to continue its calculations in the full rate for regular updates and a smooth output and it will use the last known stick position given by `_sticks.getPosition()` for that but there needs to be a timeout in some way that stops it from using stale stick positions. Before the last new commit that was only when `manual_control_setpoint` contains NANs but missing messages is the interface we use. So before the change if you kept the stick up (fly up) in auto land and lost RC in that moment it would happily continue to use the stale `getPosition()` result because it never gets updated or invalidated anymore.

> If you still want to have a check for now we can use vehicle_status.rc_signal_lost.

That works.